### PR TITLE
Sort params on an action before toString, use Params inside Crop action

### PR DESCRIPTION
**Pull request by patrick-tolosa**

### CODE REVIEW

File: src/actions/Action.ts [hunk 1]
```diff
@@ -6,18 +6,30 @@ class Action implements IAction {
   // We're using map, to overwrite existing keys. for example:
   // addParam(w_100).addParam(w_200) should result in w_200. and not w_100,w_200
   params: Map<string, Param> = new Map();
-  delimiter = ',' // {param}{delimiter}{param} for example: `${'w_100'}${','}${'c_fill'}`
+  delimiter = ','; // {param}{delimiter}{param} for example: `${'w_100'}${','}${'c_fill'}`
 
-  toString() {
-    return Array.from(this.params.values()).join(this.delimiter);
+  getSortedParams(): Param[] {
```
Comment (maoznir) [inline src/actions/Action.ts:11]: Future improvement: this and any "util" function should be static or imported from utils

File: src/actions/resize/crop.ts [hunk 2]
```diff
@@ -1,28 +1,21 @@
 import Action from "../Action";
```
Comment (maoznir) [inline src/actions/resize/crop.ts:1]: I don't think there should be a crop action, crop_mode is a param of Resize

Comment (patrick-tolosa) [reply src/actions/resize/crop.ts:1]: At this point, I just refactored the existing crop to use Params,
However, I can see how in the very near future we'll join all Resize into a single class.

We'll do it on a separate PR as it'll change a few files.

File: src/actions/Action.ts (hunk 1 above)
Comment (patrick-tolosa) [reply src/actions/Action.ts:11]: I'm even tempted to somehow have `params.sort()`  (we currently use a map, but maybe we could use our own wrapper over it)

Comment (maoznir) [reply src/actions/Action.ts:11]: yeah good idea, lets merge and refactor

**Pull request merged**